### PR TITLE
Aprilfools2022 

### PR DESCRIPTION
**Pull request by Raemon**

### CODE REVIEW

File: packages/lesswrong/components/users/UsersNameDisplay.tsx [hunk 1]
```diff
@@ -51,12 +52,30 @@ const UsersNameDisplay = ({user, nofollow=false, simple=false, classes, tooltipP
 }) => {
   const {eventHandlers} = useHover({pageElementContext: "linkPreview",  pageSubElementContext: "userNameDisplay", userId: user?._id})
 
+  const {results} = useMulti({
```
Comment (jimrandomh) [inline packages/lesswrong/components/users/UsersNameDisplay.tsx:55]: Not sure whether the performance of this query is okay. `UsersNameDisplay` has a *lot* of copies on the front page and on post pages, and doesn't normally involve a query like this.

Comment (jimrandomh) [reply packages/lesswrong/components/users/UsersNameDisplay.tsx:55]: The obvious speedup would be to move the query into a resolver, `goodheartRank`, and have the resolver use loaders to ensure the top-15 is only queried once.

File: packages/lesswrong/server/callbacks/votingCallbacks.ts [hunk 2]
```diff
@@ -12,17 +12,32 @@ import { batchUpdateScore } from '../updateScores';
...
+const hasCreatedAt = (document: any) : document is HasCreatedAtType => {
+  if (document.createdAt) return true
+  return false 
+}
+
+voteCallbacks.castVoteAsync.add(async function updateKarma({newDocument, vote}: VoteDocTuple, collection: CollectionBase<DbVoteableType>, user: DbUser) {
   // only update karma is the operation isn't done by the item's author
   if (newDocument.userId !== vote.userId && collectionsThatAffectKarma.includes(vote.collectionName)) {
     void Users.rawUpdate({_id: newDocument.userId}, {$inc: {"karma": vote.power}});
+    if (activateGoodHeartTokens && hasCreatedAt(newDocument) && newDocument.createdAt > goodHeartStartDate) {
```
Comment (jimrandomh) [inline packages/lesswrong/server/callbacks/votingCallbacks.ts:28]: Needs to check whether the voter is an established account

File: packages/lesswrong/components/users/UsersNameDisplay.tsx (hunk 1 above)
Comment (Discordius) [reply packages/lesswrong/components/users/UsersNameDisplay.tsx:55]: Yeah, that was one of the things I wanted to check tonight. Alternatively we can add either a denormalized field that saves the rank, or we can have a custom resolver, or we can pass down some props, maybe with context. Just all a bit more annoying. Was planning to do some profiling in the next half hour.

**Review by Discordius — APPROVED** [review]

**Pull request merged**